pkg/printers/internalversion/printers.go

```diff
@@ -2996,3 +3015,15 @@ func (list SortableResourceNames) Swap(i, j int) {
 func (list SortableResourceNames) Less(i, j int) bool {
 	return list[i] < list[j]
 }
+
+func isSidecarContainer(initContainer *api.Container) bool {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.SidecarContainers) {
+		return false
+	}
```

Comment:
I might not be thinking about this right, but is this needed?  Is there a reason to change how sidecar status is printed after feature gate disablement?

Reply:
Maybe I am wrong, but from my understanding, when we implement the feature gate, all changes related to the feature should be feature-gated and we should try not to change anything by default to minimize the side effects as the new feature is inevitably unstable.
Correct me if I am wrong :)

Reply:
When it is possible to completely seal off a feature via a feature gate, then I agree that is ideal to do so.

For features like this, where pods may exist in the system that depend on the feature, even after the feature is disabled, the situation is more complex.

A good goal might be that once the feature is disabled _and_ all sidecar pods are deleted, that the feature becomes fully sealed off.

For this specific codepath, while a cluster administrator is in the process of fully disabling the feature but some sidecar pods still exist, is there any value to the cluster administrator to providing accurate printer data here?

Reply:
> For features like this, where sidecar pods may continue exist in the system, even after the feature is disabled on the kube-apiserver, the situation is more complex.
> 
> A good goal might be that once the feature is disabled _and_ all sidecar pods are deleted, that the feature becomes fully sealed off.

Yes, that makes sense. I'll make sure apiserver prints the sidecar pod information correctly as far as the pod exists.

Reply:
done.